pkgs/tools/package-management/lix/common.nix

```diff
@@ -190,7 +190,7 @@ stdenv.mkDerivation {
     [
       # Enable LTO, since it improves eval performance a fair amount
       # LTO is disabled on static due to strange linking errors
-      (lib.mesonBool "b_lto" (!stdenv.hostPlatform.isStatic))
+      (lib.mesonBool "b_lto" (!stdenv.hostPlatform.isStatic && stdenv.cc.isGNU))
```

Comment:
Can we disable it for Darwin instead?

Clang LTO on Linux should be fine.

Reply:
Have you tested it? I would suspect it is more likely to be LLVM exposing UB than a Darwin issue. And the Nix derivation uses `isGNU` here.

Reply:
Tried building lix from master with clangStdenv and it built fine.

Reply:
Interesting. What about with `libc++`?

Reply:
This does work if that's what you are asking:
```
❯  nix build --impure --expr '(builtins.getFlake "github:nixos/nixpkgs/master").legacyPackages."${builtins.currentSystem}".libcxx.override {stdenv = (builtins.getFlake "github:nixos/nixpkgs/master").legacyPackages."${builtins.currentSystem}".clangStdenv;}' --print-out-paths 
/nix/store/nqpv4xbh5zqnvfn80702gahiklxxjcr3-libcxx-18.1.8

```

PS: if you have a shorthand for these overrides in cli, let me know!

Reply:
```
nix-build -E 'with import ./.{}; lix.override { stdenv = llvmPackages.libcxxStdenv; }'
```
uses `libc++`.

Reply:
How that is a problem.
```
nix build --impure --expr '(builtins.getFlake "github:nixos/nixpkgs/master").legacyPackages."${builtins.currentSystem}".lix.override {stdenv = (builtins.getFlake "github:nixos/nixpkgs/master").legacyPackages."${builtins.currentSystem}".llvmPackages.libcxxStdenv;}'
```

```log
[135/486] clang++ -Isrc/libstore/liblixstore.so.p -Isrc/libstore -I../src/libstore -Isrc/libutil -I../src/libutil -flto -fdiagnos>
FAILED: src/libstore/liblixstore.so.p/platform_linux.cc.o 
clang++ -Isrc/libstore/liblixstore.so.p -Isrc/libstore -I../src/libstore -Isrc/libutil -I../src/libutil -flto -fdiagnostics-color>
../src/libstore/platform/linux.cc:743:17: error: use of undeclared identifier 'SYS_seccomp'
  743 |     if (syscall(SYS_seccomp, SECCOMP_SET_MODE_FILTER, 0, &fprog) != 0)
      |                 ^
1 error generated.
```

Reply:
what in the hell. maybe we have a missing include but I'm guessing those sys constants are just broken. (though libcxxStdenv is SURELY not using an alternative libc right? just libc++?)

it may also be broken toolchain on the nixpkgs side for all i know, like this release blocker for lix 2.92: https://github.com/NixOS/nixpkgs/issues/177129

Reply:
This does seem to only be a missing include. Sent a patch upstream in https://gerrit.lix.systems/c/lix/+/2286

Using this override still fails when linking with things like the aws-sdk though (probably because they're using glibc?), and being a bit more forceful with libcxx usage like so

```nix
let
  pkgs = import <nixpkgs> {
    config = {
      replaceStdenv = { pkgs }: pkgs.libcxxStdenv;
    };
    overlays = [ ];
  };
in

pkgs.lix
```

results in the always fun `error: infinite recursion encountered`, so I don't think we can determine whether this is a libcxx or Darwin issue just yet

I'm inclined to say we should change this to `stdenv.cc.libcxx != null` for now, as that would cover Darwin and the possibly broken (and currently un-evaluatable) libcxx Linux build -- which can always be revisited later

Also let me know if there's anything I can help with to get this merged. Would love to see a fix after all this time :)

Reply:
i believe there is a libcxxStdenvPackages somewhere that is a nixpkgs staged from libcxxStdenv but yeah you can't really mix libc++ and libstdc++ in one binary at least if you're linking c++ symbols between them.

if this pr is acceptable i can just like, merge it. i have commit rights.

Reply:
> i think it's more important that lix works than that it has LTO in all configurations possible

Agreed.

Reply:
> if this pr is acceptable

I'd say it 100% is. My suggestions were by no means a blocker, just a way to hopefully get things going

> i believe there is a libcxxStdenvPackages somewhere that is a nixpkgs staged from libcxxStdenv

TIL that is `pkgsLLVM`...I thought that used clangStdenv. Time to build :)
![image](https://github.com/user-attachments/assets/9e83717e-be48-4122-8bc3-468a0334802d)